plugins/woocommerce-admin/client/products/layout/product-form-layout.tsx

```diff
@@ -1,8 +1,60 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Children, useEffect } from '@wordpress/element';
+import { TabPanel } from '@wordpress/components';
+
 /**
  * Internal dependencies
  */
 import './product-form-layout.scss';
 
-export const ProductFormLayout: React.FC = ( { children } ) => {
-	return <div className="product-form-layout">{ children }</div>;
+export const ProductFormLayout: React.FC< {
+	children: JSX.Element | JSX.Element[];
+} > = ( { children } ) => {
+	useEffect( () => {
+		window.document.body.classList.add(
+			'woocommerce-admin-product-layout'
+		);
+
+		return () => {
+			window.document.body.classList.remove(
+				'woocommerce-admin-product-layout'
+			);
+		};
+	}, [] );
+
+	const tabs = Children.map( children, ( child: JSX.Element ) => {
```

Comment:
I opted to extract the tabs from children so we can avoid using `applyFilters` and create the necessary SlotFills for extensibility going forward.

See the `ProductFormTab` below for usage and how a plugin might extend or create a new tab.